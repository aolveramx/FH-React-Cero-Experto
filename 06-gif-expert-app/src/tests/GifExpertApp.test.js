import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test in GifExpertApp.js', () => {

  test('should render successful', () => {
    
    const wrapper = shallow(<GifExpertApp />)

    expect(wrapper).toMatchSnapshot()

  });

  test('should render list categories', () => {
    
    const categories = ['Ninja Gayden', 'Dragon Ball']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  });
  
})