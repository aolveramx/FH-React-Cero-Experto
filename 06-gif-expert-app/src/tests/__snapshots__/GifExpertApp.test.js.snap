// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp.js should render list categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Ninja Gayden"
      key="Ninja Gayden"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test in GifExpertApp.js should render successful 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
